rstudio/rstudio
# Only close current main menu level when hitting ESC key

### Intent

- Fixes #6327
- Makes RStudio Server main menu behave more like a Windows desktop application main menu

### Approach

- Required adding a custom option in GWT's MenuBar class, see https://github.com/rstudio/gwt/pull/6
- This new behavior is off by default (opt-in) and only using it on the main menu
- Removed code that deselected the top-level menu button (e.g. "File") when closing it's popup; this was code from initial RStudio commit in 2010 and was working around an issue in older version of GWT and I could not find a case where it is still necessary

### QA Notes

- On RStudio Server, use the main menu via both keyboard shortcuts (e.g. <kbd>Alt+Shift+F</kbd> on Windows/Linux or <kbd>Ctrl+Opt+F</kbd> on Mac, move around with arrows, then hit ESC key
- The current menu level should then close, selecting the parent menu
- Try this in cases where a menu has submenus, too
- Note that the existing problem of where focus goes after you close the entire menu via ESC is covered by a separate issue, #6197
- Also do some checking around behaviors where the menu is displayed, and you click elsewhere in the RStudio UI with the mouse; make sure the menus close and are no longer showing as selected (e.g. the "File" button isn't highlighted)
- Only RStudio Server is impacted by these code changes, not desktop
